sdk/core/core-http/lib/policies/proxyPolicy.ts

```diff
@@ -13,24 +13,104 @@ import { WebResource } from "../webResource";
 import { Constants } from "../util/constants";
 import { URLBuilder } from "../url";
 
+function environmentProxyValue(name: string): string | undefined {
+  if (process.env[name]) {
+    return process.env[name];
+  } else if (process.env[name.toLowerCase()]) {
+    return process.env[name.toLowerCase()];
+  }
+  return undefined;
+}
+
 function loadEnvironmentProxyValue(): string | undefined {
+  let httpProxy = null;
+  let httpsProxy = null;
+  let allProxy = null;
+
   if (!process) {
     return undefined;
   }
 
-  if (process.env[Constants.HTTPS_PROXY]) {
-    return process.env[Constants.HTTPS_PROXY];
-  } else if (process.env[Constants.HTTPS_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTPS_PROXY.toLowerCase()];
-  } else if (process.env[Constants.HTTP_PROXY]) {
-    return process.env[Constants.HTTP_PROXY];
-  } else if (process.env[Constants.HTTP_PROXY.toLowerCase()]) {
-    return process.env[Constants.HTTP_PROXY.toLowerCase()];
+  httpsProxy = environmentProxyValue(Constants.HTTPS_PROXY)!;
+  httpProxy = environmentProxyValue(Constants.HTTP_PROXY)!;
+
+  if (httpProxy == null || httpsProxy == null) {
+    allProxy = environmentProxyValue(Constants.ALL_PROXY)!;
+
+    if (httpProxy == null) {
+      httpProxy = allProxy;
+    } else {
+      httpsProxy = allProxy;
+    }
+  }
+
+  if (httpProxy == null && httpsProxy == null) {
+    return undefined;
+  }
+
+  if (httpsProxy) {
+    return httpsProxy;
+  } else if (httpProxy) {
```

Comment:
if you declare these values without setting them to null, you could just do something simpler like

```ts
if (httpsProxy) {
  return httpsProxy;
} else {
  return httpProxy;
}
```

or even simpler:

```ts
return httpsProxy || httpProxy;
```